website/for_teachers.py

```diff
@@ -1123,26 +1145,71 @@ def parse_preview_adventure(self):
             return gettext("something_went_wrong_keyword_parsing"), 400
         return {"code": code}, 200
 
-    @route("/create-adventure", methods=["POST"])
+    def add_adventure_to_class_level(self, user, class_id, adventure_id, level, remove):
+        Class = self.db.get_class(class_id)
+        if not Class or (not utils.can_edit_class(user, Class) and not is_admin(user)):
+            return utils.error_page(error=404, ui_message=gettext("no_such_class"))
+
+        customizations, adventures, adventure_names, _, _ = self.get_class_info(
+            user, class_id)
+
+        teacher_adventures = list(self.db.get_teacher_adventures(user["username"]))
+        second_teacher_adventures = self.db.get_second_teacher_adventures([Class], user["username"])
+        teacher_adventures += second_teacher_adventures
+        is_teacher_adventure = self.is_adventure_from_teacher(adventure_id, teacher_adventures)
```

Comment:
you don't need this code since we know that this is an actual teacher's adventure!

Reply:
it also happens in `for-teachers/add-adventure`, or what is the reason why it's needed there?

Reply:
Yes, but that's a different thing. Your `add_adventure_to_class_level` function always gets teachers' adventures!

Reply:
Solved it!